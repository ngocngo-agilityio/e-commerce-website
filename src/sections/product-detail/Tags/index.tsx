// Libs
import { Flex, Text } from '@chakra-ui/react';

// Apis
import { getTagList } from '@apis';

// Utils
import { getNamesFromList } from '@utils';

interface Props {
  tagIds: number[];
}

const Tags = async ({ tagIds }: Props): Promise<JSX.Element> => {
  // Fetch data for tags
  const { data: tagList = [] } = await getTagList({ ids: tagIds });

  const tags = getNamesFromList(tagList);

  return (
    <Flex>
      <Text
        as="span"
        size="md"
        lineHeight="24px"
        color="productDetail.categoryTitle"
      >
        Tags:&nbsp;
      </Text>
      <Text size="md" lineHeight="24px">
        {tags}
      </Text>
    </Flex>
  );
};

export default Tags;
