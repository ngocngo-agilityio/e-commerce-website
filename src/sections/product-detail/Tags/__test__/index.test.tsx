// Libs
import { render } from '@testing-library/react';

// Sections
import Tags from '..';

// Mocks
import { MOCK_TAGS } from '@mocks';

const mockGetTagList = jest.fn();

jest.mock('@apis', () => ({
  ...jest.requireActual('@apis'),
  getTagList: () => mockGetTagList(),
}));

describe('Tags section', () => {
  test('should match with snapshot', async () => {
    const mockCategoryIDs = MOCK_TAGS.map((item) => +item.id);
    mockGetTagList.mockResolvedValue({ data: MOCK_TAGS });

    const { container } = render(await Tags({ tagIds: mockCategoryIDs }));

    expect(container).toMatchSnapshot();
  });
});
