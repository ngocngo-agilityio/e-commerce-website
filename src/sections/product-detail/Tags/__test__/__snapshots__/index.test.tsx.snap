// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags section should match with snapshot 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <span
      class="chakra-text css-1c9wxup"
    >
      Tags:Â 
    </span>
    <p
      class="chakra-text css-1wsccn1"
    >
      Modern, Classic, Design, Cotton
    </p>
  </div>
</div>
`;
